pyproj/crs.py

```diff
@@ -249,59 +283,100 @@ def from_epsg(cls, code):
         Parameters
         ----------
         code : int or str
-            An EPSG code. Strings will be converted to integers.
+            An EPSG code.
+
+        Returns
+        -------
+        CRS
+        """
+        return cls(_from_epsg(code))
+
+    @classmethod
+    def from_proj4(cls, in_proj_string):
+        """Make a CRS from a PROJ string
 
-        Notes
-        -----
-        The input code is not validated against an EPSG database.
+        Parameters
+        ----------
+        in_proj_string : str
+            A PROJ string.
 
         Returns
         -------
         CRS
         """
-        return cls.from_authority("epsg", code)
+        if is_wkt(in_proj_string) or "=" not in in_proj_string:
+            raise CRSError("Invalid PROJ string: {}".format(in_proj_string))
+        return cls(_from_string(in_proj_string))
 
     @classmethod
-    def from_string(cls, proj_string):
-        """Make a CRS from an EPSG, PROJ, or WKT string
+    def from_wkt(cls, in_wkt_string):
+        """Make a CRS from a WKT string
 
         Parameters
         ----------
-        proj_string : str
+        in_wkt_string : str
+            A WKT string.
+
+        Returns
+        -------
+        CRS
+        """
+        if not is_wkt(in_wkt_string):
+            raise CRSError("Invalid WKT string: {}".format(in_wkt_string))
+        return cls(_from_string(in_wkt_string))
+
+    @classmethod
+    def from_string(cls, in_crs_string):
```

Comment:
Do we still need this? It seems a bit redundant with `__init__` and `from_user_input`? 
(although it gives some kind of consistency with `to_string` ..)

Reply:
Yeah, the consistency is what I was going after here.